File: atcoder/abc350a_past_abcs.py
# vicfred
# https://atcoder.jp/contests/abc350/tasks/abc350_a
# brute force
contests = set()
for x in range(1, 350):
    contest = "ABC{:03d}".format(x)
    contests.add(contest)
contests.remove("ABC316")
ins = input()
if ins in contests:
    print("Yes")
else:
    print("No")

File: atcoder/abc350a_past_abcs.hs
-- vicfred
-- https://atcoder.jp/contests/abc350/tasks/abc350_a
-- implementation
import Data.List ((\\))

main :: IO ()
main = do
  let constest = ((++) <$> ["ABC"] <*> [replicate (3 - length (show n)) '0' ++ show n | n <- [1..349]]) \\ ["ABC316"]
  s <- getLine
  if s `elem` constest then putStrLn "Yes" else putStrLn "No"
